### PR TITLE
More fixes

### PR DESCRIPTION
**Pull request by magnet**
This brings a bunch of fixes to match flatc's behavior.

Up next, I will generate companion enums into another namespace to avoid type clashes.  In 00f4a91 namespaces can be defined easily by just declaring a `QualifiedType` but code-gen is still assuming the enum companion is in the same namespace as the union, and needs to be fixed.

### CODE REVIEW

File: butte-build/src/ir/keywords.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+pub fn substitute_keyword(input: &str) -> Option<&'static str> {
+    Some(match input {
+        "as" => "as_",
```
Comment (lovesegfault) [inline butte-build/src/ir/keywords.rs:4]: sort this

Comment (magnet) [reply butte-build/src/ir/keywords.rs:4]: That list is sorted, but there are two sublists, current keywords and reserved keywords, though I just noticed the list is outdated and does not include edition 2018.
Fixed by updating the list and adding a comment.

**Review by lovesegfault — APPROVED** [review]
Awesome!

I’m left wondering what happens in the edge case where someone has both `$keyword` and `_$keyword` in their flatbuffer definitions. I imagine things are going to explode?

Maybe `_$keyword` should be mapped to `__$keyword` as well, and so on?

**Comment by magnet** [comment]
> I’m left wondering what happens in the edge case where someone has both `$keyword` and `_$keyword` in their flatbuffer definitions. I imagine things are going to explode?

That person would really be looking for trouble :smile:.  I'd let things explode personally, though we should likely explode at the IR layer before code-gen.  That's an easy check to add ("all field names are different") but can be done later, it's an easy starter ticket if someone is interested in contributing (there was someone on Discord)

**Comment by lovesegfault** [comment]
@magnet Sweet, I'll merge this please add that ticket when you have time :)

**Pull request merged**